synthesizer/src/block/transactions/mod.rs

```diff
@@ -64,87 +66,78 @@ impl<'a, N: Network> FromIterator<&'a Transaction<N>> for Transactions<N> {
 impl<N: Network> Transactions<N> {
     /// Returns `true` if the transactions contains the given transition ID.
     pub fn contains_transition(&self, transition_id: &N::TransitionID) -> bool {
-        match cfg!(feature = "parallel") {
-            true => self.par_values().any(|tx| tx.contains_transition(transition_id)),
-            false => self.values().any(|tx| tx.contains_transition(transition_id)),
-        }
+        cfg_values!(self).any(|tx| tx.contains_transition(transition_id))
     }
 
     /// Returns `true` if the transactions contains the given serial number.
     pub fn contains_serial_number(&self, serial_number: &Field<N>) -> bool {
-        match cfg!(feature = "parallel") {
-            true => self.par_values().any(|tx| tx.contains_serial_number(serial_number)),
-            false => self.values().any(|tx| tx.contains_serial_number(serial_number)),
-        }
+        cfg_values!(self).any(|tx| tx.contains_serial_number(serial_number))
     }
 
     /// Returns `true` if the transactions contains the given commitment.
     pub fn contains_commitment(&self, commitment: &Field<N>) -> bool {
-        match cfg!(feature = "parallel") {
-            true => self.par_values().any(|tx| tx.contains_commitment(commitment)),
-            false => self.values().any(|tx| tx.contains_commitment(commitment)),
-        }
+        cfg_values!(self).any(|tx| tx.contains_commitment(commitment))
     }
 }
 
 impl<N: Network> Transactions<N> {
     /// Returns the transaction with the given transition ID, if it exists.
     pub fn find_transaction_for_transition_id(&self, transition_id: &N::TransitionID) -> Option<&Transaction<N>> {
-        match cfg!(feature = "parallel") {
-            true => self.par_values().find_any(|tx| tx.contains_transition(transition_id)),
-            false => self.values().find(|tx| tx.contains_transition(transition_id)),
-        }
+        #[cfg(not(feature = "serial"))]
+        return self.par_values().find_any(|tx| tx.contains_transition(transition_id));
+        #[cfg(feature = "serial")]
+        self.values().find(|tx| tx.contains_transition(transition_id))
```

Comment:
The match case didn't work when the serial feature was compiled because at the top of this file, rayon is conditionally disabled. Thus, when using the match case here, it doesn't disable the code in the match statement leading the compiler complaining about methods not being available. I'll give it another pass to beautify it.

Reply:
Refactored it into macros